MultiIndex stub lacks dtypes property

The stub from `MultiIndex` is missing the `dtypes` property.

**To Reproduce**

Runing `mypy` on a file with the following contents

```python
import pandas as pd

pd.MultiIndex.from_tuples([(1, 2.), (2, 3.)], names=['foo', 'bar']).dtypes
```

produces the error message ` error: "MultiIndex" has no attribute "dtypes"; maybe "dtype"?  [attr-defined]`, although `MultiIndex.dtypes` [is documented](https://pandas.pydata.org/docs/reference/api/pandas.MultiIndex.dtypes.html).

 - OS: Ubuntu 2020.04
 - python version: 3.11
 - mypy 1.1.1
 - pandas-stubs 1.5.3.230321

--- a/pandas-stubs/_typing.pyi
+++ b/pandas-stubs/_typing.pyi
@@ -306,6 +306,7 @@ S1 = TypeVar(
     int,
     float,
     complex,
+    Dtype,
     Timestamp,
     Timedelta,
     Period,

--- a/pandas-stubs/core/indexes/multi.pyi
+++ b/pandas-stubs/core/indexes/multi.pyi
@@ -4,7 +4,6 @@ from collections.abc import (
     Sequence,
 )
 from typing import (
-    Any,
     Literal,
     overload,
 )
@@ -16,6 +15,7 @@ from typing_extensions import Self
 
 from pandas._typing import (
     T1,
+    Dtype,
     DtypeArg,
     HashableT,
     np_ndarray_anyint,
@@ -70,7 +70,7 @@ class MultiIndex(Index):
     @property
     def dtype(self) -> np.dtype: ...
     @property
-    def dtypes(self) -> pd.Series[Any]: ...
+    def dtypes(self) -> pd.Series[Dtype]: ...
     def memory_usage(self, deep: bool = ...) -> int: ...
     @property
     def nbytes(self) -> int: ...

--- a/tests/test_indexes.py
+++ b/tests/test_indexes.py
@@ -15,6 +15,7 @@ from typing_extensions import (
     assert_type,
 )
 
+from pandas._typing import Dtype  # noqa: F401
 from pandas._typing import Scalar
 
 from tests import (
@@ -859,4 +860,4 @@ def test_getitem() -> None:
 def test_multiindex_dtypes():
     # GH-597
     mi = pd.MultiIndex.from_tuples([(1, 2.0), (2, 3.0)], names=["foo", "bar"])
-    check(assert_type(mi.dtypes, pd.Series), pd.Series)
+    check(assert_type(mi.dtypes, "pd.Series[Dtype]"), pd.Series)

